driver-core/src/main/java/com/datastax/driver/core/RequestHandler.java

```diff
@@ -261,12 +264,18 @@ private RetryPolicy retryPolicy() {
                 logger.trace("[{}] Starting", id);
         }
 
-        void sendRequest() {
+        void findNextHostAndQuery() {
             try {
                 Host host;
                 while (!isDone.get() && (host = queryPlan.next()) != null && !queryStateRef.get().isCancelled()) {
+
+                    HostConnectionPool pool = manager.pools.get(host);
```

Comment:
What is the motivation for checking the pool here _and_ inside the `query` method?